File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int st = 0, end = numbers.size()-1;
        vector<int> res;
        while(st < end) {
            int sum = numbers[st] + numbers[end];
            if (sum < target) 
                st += 1 ;
            else if (sum > target)
                end -= 1;
            else if (sum == target){
                res.push_back(st+1);
                res.push_back(end+1);
                return res;
            }
                
        }
        
        return res;
    }
};

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int arr[]=new int[2];
        HashMap <Integer,Integer> h1=new HashMap<>();
        for(int i=0;i<numbers.length;i++){
            int res=target-numbers[i];
            if(h1.containsKey(res)){
                arr[1]=h1.get(res);
                arr[0]=i+1;
                Arrays.sort(arr);
                return arr;
            }
            else{
                h1.put(numbers[i],i+1);
            }
        }
        return arr;
    }
}